content/sips/sip-366.md

```diff
@@ -16,36 +16,277 @@ proposal: >-
-* `delegateCollateral` can call `requestDelegateCollateral` or `processDelegateCollateral` as appropriate if collateral is being reduced and `delegateDelay` or `undelegateDelay` is set.
+### `MarketManagerModule` & `IMarketManagerModule`
+
+#### Configuration
+
+Replace logic and state associated with the minimum delegation time by omitting `minDelegationTime` and its respective setter method (`setMarketMinDelegateTime`) with the following logic and state associated with the delegation delay and window functionality:
+
+1.  `unit256 undelegateCollateralDelay` and its respective setter method `setUndelegateCollateralDelay`
+2.  `unit256 undelegateCollateralWindow` and its respective setter method `setUndelegateCollateralWindow`
+3.  `unit256 delegateCollateralDelay` and its respective setter method `setDelegateCollateralDelay`
+4.  `unit256 delegateCollateralWindow` and its respective setter method `setDelegateCollateralWindow`
+
+All of these values should be configurable only by the market.
+
+### `Vault` & `IVault`
+
+#### Intents
+
+Define the following object (i.e., `struct`) that defines all aspects of an **intent** to _delegate_ or _undelegate_ collateral. This object should closely resemble the following `DelegateCollateralIntent` object:
+
+```solidity
+/**
+ * Intent Lifecycle:
+ *
+ *          |<---- Delay ---->|<-- Processing Window -->|
+ * Time ----|-----------------|-------------------------|---->
+ *          ^                 ^                         ^
+ *          |                 |                         |
+ *  declarationTime    processingStartTime    processingEndTime
+ *
+ * Key:
+ * - declarationTime: Timestamp at which the intent is declared.
+ * - processingStartTime: Timestamp from which the intent can start being processed.
+ * - processingEndTime: Timestamp after which the intent cannot be processed.
+ *
+ * The intent can be processed only between processingStartTime and processingEndTime.
+ */
+struct DelegateCollateralIntent {
+    /**
+     * @notice An incrementing nonce to ensure the
+     * uniqueness of the intent and prevent replay attacks
+     */
+    uint256 nonce;
+    /**
+     * @notice The ID of the pool for which the account has an
+     * outstanding intent to delegate a new amount of collateral to
+     */
+    uint128 poolId;
+    /**
+     * @notice The address of the collateral type that the account has an
+     * outstanding intent to delegate a new amount of
+     */
+    address collateralType;
+    /**
+     * @notice The amount of collateral that the account has an
+     * outstanding intent to delegate a new amount of to the pool,
+     * denominated with 18 decimals of precision
+     */
+    uint256 collateralAmountD18;
+    /**
+     * @notice The intended amount of leverage associated with the new
```

Comment:
It is just a placeholder for now, and it doesn't really matter what we set this right? Shouldn't we reflect that in the coment?

Reply:
done